packages/lib/services/commands/stateToWhenClauseContext.test.ts

```diff
@@ -0,0 +1,74 @@
+import { State } from '../../reducer';
+import { getTrashFolderId } from '../trash';
+import stateToWhenClauseContext from './stateToWhenClauseContext';
+
+
+describe('stateToWhenClauseContext', () => {
+
+	it('should be inTrash if selectedNote has deleted_time and selectedFolderId is trash', async () => {
+		const appState = {
+			selectedNoteIds: ['1'],
+			selectedFolderId: getTrashFolderId(),
+			notes: [
+				{ id: '1', deleted_time: 1722567036580 },
+			],
+			folders: [],
+		} as State;
+		const result = stateToWhenClauseContext(appState);
+
+		expect(result.inTrash).toBe(true);
+	});
+
+	it('should NOT be inTrash if selectedNote deleted_time is 0', async () => {
+		const appState = {
+			selectedNoteIds: ['1'],
+			selectedFolderId: getTrashFolderId(),
+			notes: [
+				{ id: '1', deleted_time: 0 },
+			],
+			folders: [],
+		} as State;
+		const result = stateToWhenClauseContext(appState);
+
+		expect(result.inTrash).toBe(false);
+	});
+
+	it('should NOT be inTrash if selectedFolderId is not trash', async () => {
+		const appState = {
+			selectedNoteIds: ['1'],
+			selectedFolderId: 'any-other-folder',
+			notes: [
+				{ id: '1', deleted_time: 1722567036580 },
+			],
+			folders: [],
+		} as State;
+		const result = stateToWhenClauseContext(appState);
+
+		expect(result.inTrash).toBe(false);
+	});
+
+	it('should be inTrash if commandFolder has deleted_time', async () => {
+		const appState = {
+			notes: [],
+			folders: [
+				{ id: '1', deleted_time: 1722567036580, share_id: '', parent_id: '' },
+			],
+		} as State;
+		const result = stateToWhenClauseContext(appState, { commandFolderId: '1' });
+
+		expect(result.inTrash).toBe(true);
+	});
+
+	it('should NOT be inTrash if commandFolder deleted_time is 0', async () => {
+		const appState = {
+			notes: [],
+			folders: [
+				{ id: '1', deleted_time: 0, share_id: '', parent_id: '' },
+			],
+		} as State;
+		const result = stateToWhenClauseContext(appState, { commandFolderId: '1' });
+
+		expect(result.inTrash).toBe(false);
+	});
+
+});
```

Comment:
Please use plain English for the titles. Don't use variable names or at least try to avoid this - this is meant to be a high level description of the test. It should show your intention which hopefully can be expressed in English. For example "deleted_time" is "0" means note is deleted, "selectedFolderId is not trash" means the selected folder is not the trash, etc.